core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureChecker.scala

```diff
@@ -95,8 +96,16 @@ trait TransactionSignatureChecker {
       TransactionSignatureSerializer.hashForSignature(txSigComponent,
                                                       hashType,
                                                       taprootOptions)
-    val result = pubKey.verify(hash, schnorrSignature)
-    if (result) ScriptOk else ScriptErrorSchnorrSig
+    //bip341 restricts valid hash types: https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#common-signature-message
+    val invalidHashType = {
+      !(hashType.byte <= 3 || (hashType.byte >= 81.toByte && hashType.byte <= 0x83.toByte))
```

Comment:
Shouldn't this be `0x81` not `81`

https://github.com/sipa/bitcoin/blob/0e2a5e448f426219a6464b9aaadcc715534114e6/src/script/interpreter.cpp#L1523